src/content-script/worker.ts

```diff
@@ -66,21 +67,23 @@ export class Worker {
   start = async (config: WorkerConfig): Promise<void> => {
     this.clearFunc = setRandomInterval(() => this.tick(config), 500, 1500);
     this.slotsInterval = setInterval(() => this.schedule(config), 300);
-    await this.storageService.setIsSearching(true);
+    await dispatchSearchStatus({type: SearchStatusType.Started})
   };
 
-  stop = async (): Promise<void> => {
+  stop = async (status: SearchStatus = {type: SearchStatusType.Stopped}): Promise<void> => {
```

Comment:
IMHO status shouldn't be a side effect of stop. You can update the status in the main function, WDUT?